api/bucket.go

```diff
@@ -1,19 +1,205 @@
 package api
 
 import (
-	"github.com/gin-gonic/gin"
-	"github.com/konveyor/tackle2-hub/model"
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
 	"io"
+	"io/ioutil"
 	"net/http"
 	"os"
+	"path"
 	pathlib "path"
+	"path/filepath"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/konveyor/tackle2-hub/model"
 )
 
 //
 // BucketHandler provides bucket management.
 type BucketHandler struct {
 }
 
+// TODO: Validate Wildcard param to not allow access dirs outside of the bucket
+func (h *BucketHandler) serveBucketGet(ctx *gin.Context, owner *model.BucketOwner) {
```

Comment:
These two methods are called from the owner APIs. Their responsibility is to decide if a single file or an archive should be uploaded/downloaded and potentialy also validation the ```Wildcard``` parameter to not escape the bucket path.